### PR TITLE
INT B-21073 add re_countries table and refactor

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

For the last twenty years I have been working on this BL item and its beautiful personality and quirks. After fighting a timeout issue likely due to so many concurrency issues happening when building addresses (and also adding countries to the db), I updated the `db-truncate` file to NOT clear the test db of `re_countries` data so we can retain that and avoid unnecessary inserts into the test database when building test data.

OKAY.

So the point of this BL item was to add a single source of truth for country information, which required a `re_countries` table with populated data. 

In addition to adding that data, we had to then refactor how we are handling country data in the backend and adding the `Country` object to the struct and then making that data accessible to the UI.

This PR does the following:
- adds migration file that adds the `re_countries` table and refactors the `addresses` table to use the `id` column of the `re_countries` in the new `country_id` column - drops the `country` column
- updates relevant Go structs
- adds/updates a buncha `Eager`/`EagerPreload` instances to use `"Addresses.Country"` now that the `Country` object holds the necessary data
- updates the `Address` definition for Swagger to only be two characters (we will handle this later on when international rolls out)
- defaults all created addresses to have a `US` country
- updates all tests (roughly 3 million tests)

### How to test

1. Run `make server_run` and see the migrations run
2. Check your database and you should now see a `re_countries` table, as well as the `addresses` table having a removed `country` column and added `country_id` column
3. Open up the application and console log some addresses
4. You should be able to see the `Country` value in the addresses object (this is not nested at the moment, not sure if it ever will be - handling this return in the model to payload conversion)
5. Check any situation/process where `Country` is important

### CODE REVIEW

File: migrations/app/schema/20241002134836_add_re_country_table_and_refactor.up.sql [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	 ('480d884c-6329-403c-a530-fe1ccc37e20a'::uuid,'YE','YEMEN','2024-09-30 15:43:11.479097','2024-09-30 15:43:11.479097'),
+	 ('e09bfbb2-8ac2-4a25-9502-7a4c57674fcb'::uuid,'YT','MAYOTTE','2024-09-30 15:43:11.479097','2024-09-30 15:43:11.479097'),
+	 ('e5e91e42-3ee5-41cf-a435-2be3cf329780'::uuid,'ZA','SOUTH AFRICA','2024-09-30 15:43:11.479097','2024-09-30 15:43:11.479097'),
+	 ('30e49a7a-7fa1-43e5-ae3e-2438d42d1295'::uuid,'ZM','ZAMBIA','2024-09-30 15:43:11.479097','2024-09-30 15:43:11.479097');
+
+-- Update the "addresses" table since it contains an existing "country" column
+ALTER TABLE addresses ADD COLUMN IF NOT EXISTS country_id uuid;
+
+-- Populate the new column with the 'US' UUID for existing records
+UPDATE addresses
```
Comment (begrohmann) [inline migrations/app/schema/20241002134836_add_re_country_table_and_refactor.up.sql:306]: Assuming there are no other addresses with non-US countries in Prod? Or any other jibberish in there?

Comment (danieljordan-caci) [reply migrations/app/schema/20241002134836_add_re_country_table_and_refactor.up.sql:306]: Idk why there would be any other non-US countries in prod. I don't have prod db access to check. @deandreJones

Comment (deandreJones) [reply migrations/app/schema/20241002134836_add_re_country_table_and_refactor.up.sql:306]: there are none- not even possible to select a non us country at this point

**Review by begrohmann — APPROVED** [review]

**Comment by TevinAdams** [comment]
I logged some addresses and noticed the country is still within the objects which is good. Its so many changes I feel we really need to depend heavily on the test for this one to make sure nothing broke. But I am seeing the country when logging on th eUI where the address is needed.

**Review by traskowskycaci — APPROVED** [review]
Looks good!

**Comment by danieljordan-caci** [comment]
Flaky tests failing

**Review by deandreJones — APPROVED** [review]

**Comment by deandreJones** [comment]
what's the worst that could happen...

**Comment by danieljordan-caci** [comment]
> what's the worst that could happen...

HOP IN HERE WE GO!

**Pull request merged**